ludwig/data/dataframe/dask.py

```diff
@@ -129,7 +133,25 @@ def to_ray_dataset(self, df):
         return from_dask(df)
 
     def from_ray_dataset(self, dataset) -> dd.DataFrame:
-        return dataset.to_dask()
+        """Custom Ray to Dask conversion implementation to pass in meta during dd.DataFrame creation."""
+
+        @dask.delayed
+        def block_to_df(block: Block):
+            block = BlockAccessor.for_block(block)
+            if isinstance(block, (ray.ObjectRef, ClientObjectRef)):
+                raise ValueError(
+                    "Dataset.to_dask() must be used with Dask-on-Ray, please "
+                    "set the Dask scheduler to ray_dask_get (located in "
+                    "ray.util.dask)."
+                )
+            return block.to_pandas()
+
+        # Use a sample of the ray dataset to generate meta to infer types even if there are NaNs
+        meta = dataset.limit(META_SAMPLE_SIZE).to_pandas()
```

Comment:
This is also a significant change. Could you add a note about this in the docstring? Does this change any inferred types for existing datasets?

Reply:
@justinxzhao I will update this PR with a docstring to make it clear why we're making this change. 
As far as I know, this doesn't break any existing tests or types for existing datasets. I haven't come across any issues during benchmarking, and I believe @geoffreyangus also ran benchmarks with this change across a variety of datasets and no issues popped up. 

@geoffreyangus are you able to confirm?